aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -592,13 +592,20 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
 
 // ALL REDUCE #################################################################
 
-static ScalarType get_dtype(Tensor& result, const Tensor& self, optional<ScalarType> dtype,
-                            bool promote_integers=false) {
+inline ScalarType get_dtype(Tensor& result, optional<ScalarType> dtype) {
+  TORCH_CHECK(result.defined(), "Cannot create a new tensor inside a reduction op. You likely tried to call an operator with an out argument but the out argument was an undefined tensor.");
```

Comment:
Sounds more like an INTERNAL_ASSERT to me :>

Reply:
This can be triggered by user code that calls the C++ API with an undefined tensor